app/services.js

```diff
@@ -0,0 +1,254 @@
+"use strict";
+
+angular.module('app')
+
+.service('configManager', function(localStorageService) {
+  var configManager = {};
+
+  configManager.getPrivateToken = function() {
+    return localStorageService.get('private_token');
+  }
+
+  configManager.setPrivateToken = function(privateToken) {
+    localStorageService.get('private_token', privateToken);
+  }
+
+  configManager.getUrl = function() {
+    return localStorageService.get('url');
+  }
+
+  configManager.setUrl = function(url) {
+    localStorageService.set('url', url);
+  }
+
+  configManager.getRefreshRate = function() {
+    return localStorageService.get('refresh_rate') || 5;
+  }
+
+  configManager.setRefreshRate = function(refreshRate) {
+    localStorageService.set('refresh_rate', refreshRate);
+  }
+
+  configManager.displayBranchColumn = function() {
+    return localStorageService.get('display_branch_column') || true;
+  }
+
+  configManager.setDisplayBranchColumn= function(displayBranchColumn) {
+    localStorageService.set('display_branch_column', displayBranchColumn);
+  }
+
+  configManager.clearAll = function() {
+    localStorageService.clearAll();
+  }
+
+  return configManager;
+})
+
+.service('gitLabManager', function(configManager, $http, $q) {
+  var gitLabManager = {};
+
+  gitLabManager.getUser = function() {
+    var deferred = $q.defer();
+
+    if (!gitLabManager.hasCredentials()) {
+      deferred.reject("Url and/or private token are missing");
+    } else {
+      $http({
+        url: configManager.getUrl() + '/api/v3/user',
+        headers:  {'PRIVATE-TOKEN': configManager.getPrivateToken()}
+      }).then(function(response) {
+        deferred.resolve(response.data);
+      }, function(msg) {
+        deferred.reject(msg);
+      });
+    }
+
+    return deferred.promise;
+  }
+
+  gitLabManager.hasCredentials = function() {
+    return configManager.getUrl() && configManager.getPrivateToken();
+  }
+
+  gitLabManager.authenticate = function(url, privateToken) {
+    configManager.setUrl(url);
+    configManager.setPrivateToken(privateToken);
+
+    var deferred = $q.defer();
+
+    gitLabManager.getUser().then(function(user) {
+      deferred.resolve(user);
+    }, function() {
+      gitLabManager.logout();
+      deferred.reject('Unauthorized');
+    });
+
+    return deferred.promise;
+  }
+
+  gitLabManager.logout = function() {
+    configManager.clearAll();
```

Comment:
Maybe we should keep the config which is not directly related to GitLab. Refresh rate and display branch column for example.